launch_test.go

```diff
@@ -1,42 +1,156 @@
 package main
 
-import (
-	"fmt"
+import(
+	"os"
+	"os/exec"
 	"strings"
 	"testing"
 )
+var baseDir string
 
-var shellOutSetup map[string]string
+// Create a test directory and change into it
+func setUp() {
+	os.Mkdir("testDir", 0777)
+	os.Chdir("testDir")
+}
+
+// Change to base dirrectory and remove the test directory
+func cleanUp() {
+	os.Chdir(baseDir)
+	ShellCommand("rm", "-rf", "testDir")
+}
 
-type shellOutMock struct{}
+// Test that a valid shell command will execute properly
+func TestShellCommand(t *testing.T) {
+	var wdByte []byte
+	wdByte, _ = exec.Command("pwd").CombinedOutput()
+	wd := string(wdByte)
 
-func (s shellOutMock) run(cmd string, args ...string) (out []byte, err error) {
-	cmdslice := []string{cmd}
-	cmdslice = append(cmdslice, args...)
+	wdTest, _ := ShellCommand("pwd")
 
-	cmd = strings.Join(cmdslice, " ")
-	output := shellOutSetup[cmd]
-	if output == "" {
-		return nil, fmt.Errorf("Error: %s not mocked", cmd)
+	// Working directories should be equal
+	if strings.Compare(wd, wdTest) != 0 {
+		t.Logf("Expected working directory to equal: %s, but was: %s", wd, wdTest)
+		t.FailNow()
 	}
+}
 
-	return []byte(output), nil
+// Test that an invalid shell command will return an error
+func TestShellCommandBadInput(t *testing.T) {
+	_, err := ShellCommand("pwda")
+
+	// Error message should not be nil
+	if err == nil {
+		t.Logf("Expected err to not be: nil")
+		t.Fail()
+	}
 }
 
-func Test(t *testing.T) {
-	shellOut = shellOutMock{}
+// Test that Clone will return without errors and change into the cloned directory
+func TestClone(t *testing.T) {
+	setUp()
+	wd, _ := ShellCommand("pwd")
 
-	shellOutSetup = map[string]string{
-		"/opt/screwdriver/launch.sh abc": "abc",
+	// Arguments
+	GIT_PATH := "github.com/Filbird/pr"
+	GIT_BRANCH := "master"
+
+	err := Clone(GIT_PATH, GIT_BRANCH)
+	// Error message should be nil
+	if err != nil {
+		t.Logf("Expected err to be: nil, but was %v", nil)
+		cleanUp()
+		t.FailNow()
 	}
 
-	output, err := runLauncher("abc")
-	if string(output) != shellOutSetup["/opt/screwdriver/launch.sh abc"] {
-		errmsg := fmt.Sprintf("Expected %s, got %s", shellOutSetup["echo"], output)
-		t.Error(errmsg)
+	wdTest, _ := ShellCommand("pwd")
+	wd = strings.Trim(wd, "\n")
+	wd = wd + "/" + GIT_PATH + "\n"
+	// Working directory should be updated
+	if strings.Compare(wd, wdTest) != 0 {
+		t.Logf("Expected working directory to equal: %s, but was: %s", wd, wdTest)
```

Comment:
You can just do t.Failf("Expected working directory to equal: %s, but was: %s", wd, wdTest)